main/src/main/java/cgeo/geocaching/CgeoApplication.java

```diff
@@ -166,4 +169,13 @@ private static String getApplicationHashkey(final Context context) {
         return sb.toString();
     }
 
+    public static void acquireUnifiedMap(final UnifiedMapActivity newUnifiedMapActivity) {
```

Comment:
Still don't like the idea...

But if you really want to store a global weak reference to a UnifiedMapActivity then you might as well do it directly in the UnifiedMapActivity class. No need to do that in CgeoApplication.

BTW: what happens when for some reason two UnifiedMapActivities exist at the same time? (this is perfectly allowed by Android lifecycle)

Reply:
What can we use instead to notify the "old" activity on demand? It should return to the caller (`UnifiedMapActivity` via means of `CGeoApplication` in this case) after it has reached the old activity (or if the old activity does no longer exist), so that the new MapView is only created after the old one got destroyed.

Having two instances of UnifiedMapActivity at the same time within the same application context is the thing we need to protect against, to my understanding, as then VTM would use the same global vars and state info, which mixes things up.
(Not sure what would happen if we start a UnifiedMap, then go back to the mobile and start another UnifiedMap using a direct link. Would this still be the same application context - then the implemented protection should help - , or if not, then VTM should not run into those problems.)

Reply:
IMHO the "on demand" part is the problem. 

Have you tested if always recreating the fragment really significantly degrese performance? As doing so would significantly streamline the code. 

(We just discussed in university that one of the reasons why architecture gets hard to maintain is caused by developers trying to optimize performance too early :) )

Reply:
> (`UnifiedMapActivity `via means of `CGeoApplication` in this case)

Again: as long as its only UnifiedMapActivity handling VTM, the code from CgeoApplication should be moved there. No need to put it into CgeoApplication. You just need a place to store a "global" weak reference, this might as well be a static variable in class `UnifiedMapActivity`.

Reply:
done

and added a visual comparison of both methods, see below